src/net/udp/tftp.c

```diff
@@ -323,12 +323,24 @@ void tftp_set_mtftp_port ( unsigned int port ) {
  * @ret rc		Return status code
  */
 static int tftp_send_rrq ( struct tftp_request *tftp ) {
-	const char *path = tftp->uri->path;
 	struct tftp_rrq *rrq;
+	const char *path;	
 	size_t len;
 	struct io_buffer *iobuf;
 	size_t blksize;
 
+	/* Strip initial '/' if present. If we were opened via the
+	* URI interface, then there will be an initial '/', since a
+	* full tftp:// URI provides no way to specify a non-absolute
+	* path. However, many TFTP servers (particularly Windows
+	* TFTP servers) complain about having an initial '/', and it
+	* violates user expectations to have a '/' silently added to
+	* the DHCP-specified filename.
+	*/
+	path = tftp->uri->path;
```

Comment:
how is lines 327 and 340 different from line 326?

I like your comment though.